controllers/project.lua

```diff
@@ -146,13 +146,8 @@ ProjectController = {
             -- Description: Get a particular project.
             --              Response will depend on query issuer permissions.
             -- Parameters:  delta, ispublic, ispublished
-            local project =
-                Projects:find(self.params.username, self.params.projectname)
-
-            if not project then yield_error(err.nonexistent_project) end
-            if not (project.ispublic or users_match(self)) then
-                assert_admin(self, err.nonexistent_project)
-            end
+            local project = assert_project(self, self.params.username, self.params.projectname)
+            assert_error(can_view_project(self, project), err.nonexistent_project)
```

Comment:
`assert_error` is built into lapis, I think we should use it more.
This seems enough without necessarily writing `assert_user_can_view_project()` but that's an option, too. 

I actually wonder if the `can_view project` method should really be `project:user_can_view(self.current_user)` and directly belong to the project model. That makes writing the assert a bit more clear.